tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystem.java

```diff
@@ -0,0 +1,80 @@
+package org.togetherjava.tjbot.tag;
+
+import org.jooq.Result;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.Tags;
+import org.togetherjava.tjbot.db.generated.tables.records.TagsRecord;
+
+import java.util.*;
+
+public class TagSystem {
+    private final Database database;
+    private boolean changed = true;
+    private Map<String, String> cache;
```

Comment:
A non-threadsafe cache sounds a bit annoying. I would just not cache at all until profiling shows it would be beneficial enough.